### PR TITLE
Add provision and teardown run_stages

### PR DESCRIPTION
**Pull request by ehankland**
https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/issues/409

### CODE REVIEW

**Comment by skschneider** [comment]
Make sure this cleanup section is appropriate for the new TEARDOWN stage:
https://github.com/ehankland/PerfKitBenchmarker/blob/e81b443b6616dec5f5f2e4ee436005aca4d48147/perfkitbenchmarker/pkb.py#L360

**Comment by ehankland** [comment]
Thanks for catching that - I think it makes sense now - PTAL

**Comment by skschneider** [comment]
I checked out your branch and tried it. Someone who used to do the prepare stage first would encounter an error message like this:

```
2015-11-19 10:33:51,830 6f691648 MainThread INFO     Verbose logging to: /tmp/perfkitbenchmarker/run_6f691648/pkb.log
2015-11-19 10:33:51,830 6f691648 MainThread INFO     PerfKitBenchmarker version: v0.23.0-180-g5b9f1da
2015-11-19 10:33:51,935 6f691648 MainThread INFO     Flag values:
--run_stage=prepare
--benchmarks=cluster_boot
2015-11-19 10:33:51,952 6f691648 MainThread cluster_boot(1/1) ERROR    Unable to unpickle spec file for benchmark cluster_boot0.
2015-11-19 10:33:51,952 6f691648 MainThread cluster_boot(1/1) ERROR    Error during benchmark cluster_boot
Traceback (most recent call last):
  File "[...]/perfkitbenchmarker/pkb.py", line 332, in RunBenchmark
    spec = benchmark_spec.BenchmarkSpec.GetSpecFromFile(benchmark_uid)
  File "[...]/perfkitbenchmarker/benchmark_spec.py", line 346, in GetSpecFromFile
    raise e
IOError: [Errno 2] No such file or directory: '/tmp/perfkitbenchmarker/run_6f691648/cluster_boot0'
2015-11-19 10:33:51,952 6f691648 MainThread cluster_boot(1/1) ERROR    Benchmark 1/1 cluster_boot (UID: cluster_boot0) failed. Execution will continue.
2015-11-19 10:33:51,953 6f691648 MainThread cluster_boot(1/1) INFO     Benchmark run statuses:
-----------------------------------
Name          UID            Status
-----------------------------------
cluster_boot  cluster_boot0  FAILED
-----------------------------------
Success rate: 0.00% (0/1)
2015-11-19 10:33:51,953 6f691648 MainThread cluster_boot(1/1) INFO     Complete logs can be found at: /tmp/perfkitbenchmarker/run_6f691648/pkb.log
2015-11-19 10:33:51,953 6f691648 MainThread cluster_boot(1/1) INFO     To run again with this setup, please use --run_uri=6f691648
```

I think that in the short term, there should be some kind of message in there that explains why this didn't work.

**Comment by ehankland** [comment]
Good point - fixed - PTAL

File: perfkitbenchmarker/pkb.py [hunk 1]
```diff
@@ -294,17 +319,29 @@ def RunBenchmark(benchmark, collector, sequence_number, total_benchmarks,
...
-          spec = benchmark_spec.BenchmarkSpec.GetSpecFromFile(benchmark_uid)
+          try:
+            spec = benchmark_spec.BenchmarkSpec.GetSpecFromFile(benchmark_uid)
+          except IOError:
+            logging.error(
+                'We were unable to load the BenchmarkSpec. This may be related '
+                'to two additional run stages which have recently been added. '
+                'Please make sure to run the stage "provision" before '
+                '"prepare". Similarly, make sure to run "teardown" after '
+                '"cleanup".')
```
Comment (skschneider) [inline perfkitbenchmarker/pkb.py]: This could probably be limited to just STAGE_PREPARE.

**Comment by skschneider** [comment]
Thanks. One comment, but otherwise LGTM.

**Comment by ehankland** [comment]
Done

**Pull request merged**